cmd/kube-apiserver/app/server.go

```diff
@@ -501,12 +502,19 @@ func buildGenericConfig(
 	}
 	versionedInformers = clientgoinformers.NewSharedInformerFactory(clientgoExternalClient, 10*time.Minute)
 
-	genericConfig.Authentication.Authenticator, genericConfig.OpenAPIConfig.SecurityDefinitions, err = BuildAuthenticator(s, genericConfig.EgressSelector, clientgoExternalClient, versionedInformers)
+	var addAuthenticatorHooks func(*genericapiserver.Config) error
+	genericConfig.Authentication.Authenticator, genericConfig.OpenAPIConfig.SecurityDefinitions, addAuthenticatorHooks, err = BuildAuthenticator(s, genericConfig.EgressSelector, clientgoExternalClient, versionedInformers)
```

Comment:
or this method could provide additional poststarthooks to add.  If you name them carefully, then you'll be de-dupe calls.